### PR TITLE
Animesuge.io | Add Anime Suge Presence

### LINKED ISSUE #3368: AnimeSuge | AnimeSuge.io
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?**  Whitzy#0001

**What's the name of the service?**  Animesuge.io

**What should the Presence display?**  What you are watching.

**If possible, please provide a logo for the service (512x512 minimum)**  Sadly I do not have one for it.

### PR DESCRIPTION
**Pull request by MMDDKK6500**
Closes #3368 
ezgif com-gif-maker (1)
AnimeSuge is a free streaming anime website that allows you to watch anime online in English subbed and dubbed.
First time i do a presence, i hope i did everything right.
_There may be some typos, I'm Brazillian and I have a limited knowledge of English_
also, I don't watch anime

### CODE REVIEW

**Comment by SlowLife1661** [comment]
Add timestamp for video

File: websites/A/Animesuge/presence.ts [hunk 1]
```diff
@@ -24,22 +24,39 @@ const presence = new Presence({
...
+  currentTime: number;
+
+function getTimestamps(
+  videoTime: number,
+  videoDuration: number
+): Array<number> {
+  const startTime = Date.now(),
+    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
+  return [Math.floor(startTime / 1000), endTime];
+}
```
Comment (Bas950) [inline websites/A/Animesuge/presence.ts:37]: there is a presence. function for this

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Animesuge/presence.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/added") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
+  } else if (page === "/tv") {
+    data.details = pages[page];
+    data.state = "Searching animes";
+    data.startTimestamp = Math.floor(Date.now() / 1000);
```
Comment (Bas950) [inline websites/A/Animesuge/presence.ts:90]: you can just make 1 timestamp and set it at the start, this is overkill doing it 9000+ times

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Slowlife01 — CHANGES_REQUESTED** [review]
Show new proof of working

**Comment by MMDDKK6500** [comment]
> Show new proof of working

ezgif com-gif-maker (3)

**Comment by SlowLife1661** [comment]
Ah what about what it's pauses

**Comment by MMDDKK6500** [comment]
> Ah what about when it pauses

ezgif com-gif-maker (4)

**Comment by Bas950** [comment]
> > Ah what about when it pauses
> 
> I used Opera GX Picture In Picture to help me record it
> 
> ezgif com-gif-maker (4)

Maybe show a play/pause icon and remove timestamp when paused.

**Comment by MMDDKK6500** [comment]
> Maybe show a play/pause icon and remove timestamp when paused.
I am on vacation right now so I won't be able to code in the next 3 days, sorry

**Comment by MMDDKK6500** [comment]
proof that it's working
ezgif com-gif-maker (6)

File: websites/A/Animesuge/presence.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  } else if (page.includes("/anime")) {
+    data.details = pages["/anime"];
+    data.state = `Watching ${animeName.textContent}`;
+    if (!paused) {
+      data.smallImageKey = "play";
+      [data.startTimestamp, data.endTimestamp] = timestamps;
+    } else {
+      data.smallImageKey = "pause";
+      data.smallImageText = "Paused";
+      [data.startTimestamp, data.endTimestamp] = [undefined, undefined];
```
Comment (Bas950) [inline websites/A/Animesuge/presence.ts:61]:
```suggestion
      delete data.startTimestamp;
      delete data.endTimestamp;
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Animesuge/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "author": {
+      "name": "MMDDKK",
+      "id": "259984855503339520"
+    },
+    "service": "AnimeSuge",
+    "altnames": ["Animesuge","Anime Suge"],
```
Comment (Bas950) [inline websites/A/Animesuge/dist/metadata.json:8]:
```suggestion
```

File: websites/A/Animesuge/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    "description": {
+      "en": "AnimeSuge is a free streaming anime website that allows you to watch anime online in English subbed and dubbed."
+    },
+    "url": "animesuge.io",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/iQWi18m.png",
```
Comment (Bas950) [inline websites/A/Animesuge/dist/metadata.json:14]: logo must be 512x512

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/Animesuge/presence.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+      [data.startTimestamp, data.endTimestamp] = timestamps;
+    } else {
+      data.smallImageKey = "pause";
+      data.smallImageText = "Paused";
+      delete data.startTimestamp;
+      delete data.endTimestamp;
+    }
+    data.buttons = [
+      {
+        label: "Watch it too!",
```
Comment (Bas950) [inline websites/A/Animesuge/presence.ts:66]:
```suggestion
        label: "Watch Episode",
```

File: websites/A/Animesuge/presence.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      startTimestamp: Math.floor(Date.now() / 1000)
+    },
+    search: URLSearchParams = new URLSearchParams(document.location.search.substring(1)),
+    timestamps = presence.getTimestamps(currentTime, timeEnd);
+  if (page in pagesSearching) {
+    data.details = pages[page];
+    data.state = "Searching animes";
+  } else if (page.includes("/anime")) {
+    data.details = pages["/anime"];
+    data.state = `Watching ${animeName.textContent}`;
```
Comment (Bas950) [inline websites/A/Animesuge/presence.ts:54]: maybe do
details = title
state = episode etc.?

**Comment by MMDDKK6500** [comment]
proof that it works
ezgif com-gif-maker (7)

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Comment by tatsumioga1** [comment]
Is it possible to change the "Playing a game" to "watching" or something? I was wondering since an extension on chromium allows us to do that (tho you need to be using discord on browser to use it unlike premid and its all manual)

**Comment by MMDDKK6500** [comment]
> Is it possible to change the "Playing a game" to "watching" or something? I was wondering since an extension on chromium allows us to do that (tho you need to be using discord on browser to use it unlike premid and its all manual)

Thanks for your suggestion, but unfortunately PreMiD is only able to use "Playing a game" because PreMiD uses Discord's Rich Presence feature that was made for games, the only exception is Spotify and Twitch(I think).
Next time you want to suggest something, please use the [Issues](https://github.com/PreMiD/Presences/issues) tab on the PreMiD Presences github repo, not the pull request for the presence, thank you

**Pull request merged**